docs/sources/administration/provisioning/index.md

```diff
@@ -53,7 +53,30 @@ datasources:
       password: $PASSWORD
 ```
 
-To escape a literal `$` in your provisioning file values, use `$$`.
+### Use of the special character `$`
+
+Grafana's provisioning system considers any set of characters after an `$` a variable name. 
+
+During the replacement process, Grafana:
+
+1. Replaces the variables that use the syntax `${ENV_VAR_NAME}`.
+1. Next, it replaces the variables that use the syntax `$ENV_VAR_NAME`. 
+
+If your data contains the character `$` and you're using an environment variable, use the `$ENV_VAR_NAME` syntax to avoid double expansion. If you use the `${ENV_VAR_NAME}` syntax, the value will be first replaced as `Pa$sw0rd` and then again as `Pa` since `$sw0rd` will be considered another variable.
```

Comment:
```suggestion
If your data contains the character `$`, for example `Pa$sw0rd`, and you're using an environment variable, use the `$ENV_VAR_NAME` syntax to avoid double expansion. If you use the `${ENV_VAR_NAME}` syntax, the value will be first replaced as `Pa$sw0rd` and then again as `Pa` since `$sw0rd` will be considered another variable.
```